src/components/calculator.js

```diff
@@ -1,41 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../App.css';
+import calculate from '../logic/calculate';
+import CalcButton from './calc-button';
 
-const Calculator = () => (
-  <section className="calc-UI">
-    <div className="keys">
-      <input type="text" id="display" />
-    </div>
-    <div className="keys">
-      <button type="button" className="calc-Btns">AC</button>
-      <button type="button" className="calc-Btns">+/-</button>
-      <button type="button" className="calc-Btns">%</button>
-      <button type="button" className="calc-Btns">&divide;</button>
-    </div>
-    <div className="keys">
-      <button type="button" className="calc-Btns">7</button>
-      <button type="button" className="calc-Btns">8</button>
-      <button type="button" className="calc-Btns">9</button>
-      <button type="button" className="calc-Btns">x</button>
-    </div>
-    <div className="keys">
-      <button type="button" className="calc-Btns">4</button>
-      <button type="button" className="calc-Btns">5</button>
-      <button type="button" className="calc-Btns">6</button>
-      <button type="button" className="calc-Btns">-</button>
-    </div>
-    <div className="keys">
-      <button type="button" className="calc-Btns">1</button>
-      <button type="button" className="calc-Btns">2</button>
-      <button type="button" className="calc-Btns">3</button>
-      <button type="button" className="calc-Btns">+</button>
-    </div>
-    <div className="keys">
-      <button type="button" className="calc-Btns">0</button>
-      <button type="button" className="calc-Btns">.</button>
-      <button type="button" className="calc-Btns">=</button>
-    </div>
-  </section>
-);
+const Calculator = () => {
+  const [calcObj, setCalcObj] = useState({ total: null, next: null, operation: null });
+
+  const handleButtonClick = (buttonName) => {
+    const newCalcObj = calculate(calcObj, buttonName);
+    setCalcObj(newCalcObj);
+  };
+
+  return (
+    <section className="calc-UI">
+      <div id="display">
+        {calcObj.next || calcObj.operation || calcObj.total || '0'}
+      </div>
+      <div className="keys">
+        <CalcButton value="AC" onClick={() => handleButtonClick('AC')} />
+        <CalcButton value="+/-" onClick={() => handleButtonClick('+/-')} />
+        <CalcButton value="%" onClick={() => handleButtonClick('%')} />
+        <CalcButton value="&divide;" onClick={() => handleButtonClick('&divide;')} />
```

Comment:
- [ ] Currently when I click on the divide button, it does not work, it would be better to use this symbol `%` or operator  `&#247;` for  division

Reply:
Thank you for the suggestion. I have modified the button and event handler in my last commit to fix the error.